### PR TITLE
Unify data and template of field resolvers and operations

### PR DESCRIPTION
**Pull request by joffrey-bion**
Conceptually, GraphQL operations (queries/mutations/subscriptions) are a
special case of field resolvers that happen to be on well known root types.

Technically, this means that everything about field resolvers is the same
as these root types resolvers, apart from the fact that the former get an
instance of the resolved type as first param.

Therefore, all the configuration that applies to field resolvers also
applies to resolvers of the fields that are part of these root types, like
CompletableFuture return type, or DataFetcherEnvironment parameter.

Resolves: https://github.com/kobylynskyi/graphql-java-codegen/issues/64
Resolves: https://github.com/kobylynskyi/graphql-java-codegen/issues/61
Resolves: https://github.com/kobylynskyi/graphql-java-codegen/issues/63

### CODE REVIEW

**Comment by joffrey-bion** [comment]
@kobylynskyi thanks for the feedback, yes the plugin changes were on the way :)

File: src/test/resources/expected-classes/CommitResolver.java.txt [hunk 1]
```diff
@@ -7,14 +7,19 @@ public interface CommitResolver {
 
     PullRequestConnection associatedPullRequests(Commit commit, String after, String before, Integer first, Integer last, PullRequestOrder orderBy, DataFetchingEnvironment env) throws Exception;
 
+    @javax.validation.constraints.NotNull
+    @com.fasterxml.jackson.databind.annotation.JsonDeserialize(using = com.example.json.DateTimeScalarDeserializer.class)
```
Comment (kobylynskyi) [inline src/test/resources/expected-classes/CommitResolver.java.txt:11]: @joffrey-bion I think its a bit late to ask but why do we have this annotation here?

**Comment by kobylynskyi** [comment]
@joffrey-bion I think we can release 1.6.0 and this will be the last changes
Please let me know and I will release tmrw morning (US time)

**Comment by joffrey-bion** [comment]
That's great news! About the annotation, it's just matching the config that was done in the test. It also bugged me at first 😄 I guess it was really just a test to check if the annotation was put correctly, not an actual use case, because I see no reason for Blame to be a date

**Pull request merged**